src/overlays/actors/ovl_En_Ani/z_en_ani.c

```diff
@@ -28,62 +44,314 @@ const ActorInit En_Ani_InitVars = {
+                         Animation_GetLastFrame(&gAniTreeHangLosingBalanceAnim), 2, -5.0f);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ani/func_80967D20.s")
+void EnAni_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnAni* this = THIS;
+    f32 minVelocity;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ani/func_80967DA0.s")
+    Collider_UpdateCylinder(&this->actor, &this->collider1);
+    Collider_UpdateCylinder(&this->actor, &this->collider2);
+    CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->collider1.base);
+    if ((this->stateFlags & ANI_STATE_UNUSED) == 0) {
+        CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->collider2.base);
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ani/func_80967DCC.s")
+    this->actor.velocity.y += this->actor.gravity;
+    minVelocity = this->actor.minVelocityY;
+    if (this->actor.velocity.y < minVelocity) {
+        this->actor.velocity.y = minVelocity;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ani/func_80967E34.s")
+    Actor_ApplyMovement(&this->actor);
+    Actor_UpdateBgCheckInfo(globalCtx, &this->actor, 0.0f, 0.0f, 0.0f, 4);
+    this->actionFunc(this, globalCtx);
+    if (this->actor.xzDistToPlayer < 100.0f && (this->stateFlags & ANI_STATE_CLIMBING) == 0) {
+        func_800E9250(globalCtx, &this->actor, &this->headRot, &this->unk2E6, this->actor.focus.pos);
+        this->unk2E6.x = this->unk2E6.y = this->unk2E6.z = 0;
+    } else {
+        Math_SmoothStepToS(&this->headRot.x, 0, 6, 6200, 100);
+        Math_SmoothStepToS(&this->headRot.y, 0, 6, 6200, 100);
+        Math_SmoothStepToS(&this->unk2E6.x, 0, 6, 6200, 100);
+        Math_SmoothStepToS(&this->unk2E6.y, 0, 6, 6200, 100);
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ani/func_80967E90.s")
+    this->blinkFunc(this);
+    if ((this->stateFlags & ANI_STATE_FALLING) != 0) {
+        if (this->actor.cutscene == -1) {
+            this->stateFlags &= ~ANI_STATE_FALLING;
+        } else if (ActorCutscene_GetCanPlayNext(this->actor.cutscene)) {
+            ActorCutscene_StartAndSetUnkLinkFields(this->actor.cutscene, &this->actor);
+            this->actor.cutscene = ActorCutscene_GetAdditionalCutscene(this->actor.cutscene);
+            func_800E02AC(Play_GetCamera(globalCtx, ActorCutscene_GetCurrentCamera(this->actor.cutscene)),
+                          &this->actor);
+        } else {
+            ActorCutscene_SetIntentToPlay(this->actor.cutscene);
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ani/func_80967F20.s")
+s32 EnAni_OverrideLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot, Actor* actor) {
+    EnAni* this = (EnAni*)actor;
+    if (limbIndex == 15) { // HEAD
+        rot->x += this->headRot.y;
+        rot->z += this->headRot.x;
+    }
+    return 0;
```

Comment:
```suggestion
    return false;
```
This is used as a boolean.